components/UserProvider.js

```diff
@@ -94,6 +97,10 @@ class UserProvider extends React.Component {
 
       // Store the error
       this.setState({ loadingLoggedInUser: false, errorLoggedInUser: error.message });
+      if (error.message.includes('Two-factor authentication is enabled')) {
```

Comment:
We should avoid testing errors on plain text strings. What happens if we change that sentence?

Reply:
How would you suggest setting the states for enforcing 2fa and so on then? You said we should throw/catch 'errors' but they aren't necessarily 'errors', just that we need to challenge the user to enter the 2FA code. So if we aren't checking the 'error' text then how else would we do this?

Reply:
On line 94 there is also already existing code that does the same thing, checking the error message